src/__test__/integration/peer.spec.ts

```diff
@@ -173,143 +163,90 @@ describe('Typescript usage suite', () => {
                 (call "${peer2.getStatus().peerId}" ("test" "test") ["test"])
             )
         `;
-        const particle = Particle.createNew(script);
-        await peer1.internals.initiateParticle(particle, doNothing);
+        const particle = peer1.internals.createNewParticle(script);
+
+        if (particle instanceof Error) {
+            throw particle;
+        }
 
-        // assert
-        const res = await resMakingPromise;
-        expect(res).toEqual('test');
+        peer1.internals.initiateParticle(particle, doNothing);
+
+        expect(await res).toEqual('test');
 
         await peer1.stop();
         await peer2.stop();
     });
 
     describe('should make connection to network', () => {
         it('address as string', async () => {
-            // arrange
-            const addr = nodes[0];
-
-            // act
-            await peer.start({ connectTo: addr });
+            await peer.start({ connectTo: nodes[0].multiaddr });
             const isConnected = await checkConnection(peer);
 
-            // assert
             expect(isConnected).toBeTruthy();
         });
 
         it('address as multiaddr', async () => {
-            // arrange
-            const addr = new Multiaddr(nodes[0].multiaddr);
-
-            // act
-            await peer.start({ connectTo: addr });
+            await peer.start({ connectTo: new Multiaddr(nodes[0].multiaddr) });
```

Comment:
I think we should either use "AAA" pattern universally or not use it at all. At least in terms of pure unit tests.

IMO (esp, coming from backend / C# background) "AAA" thing is very neat. It forces you to think about the precise scope of the unit test.

Reply:
Btw this test series can be written in table form. Seems like...

Reply:
I haven't heard about AAA before. In my personal opinion adding comments doesn't make anything clearer, I accidentally removed them. But with tests it's almost always this pattern anyways, but without explicit comments. We can discuss if it's easier for you to read, maybe. I would suggest regarding this question to do whatever you want - I don't think consistency matters here. Some tests could have the comments some test could be ok without them and we always can add or remove them later